controllers/activemqartemis_jdbc_ha_test.go

```diff
@@ -0,0 +1,386 @@
+												},
+											},
+											InitialDelaySeconds: 10,
+											PeriodSeconds:       5,
+										},
+									},
+								},
+							}},
+					},
+				}
+				Expect(k8sClient.Create(ctx, &db)).Should(Succeed())
+
+				dbService := corev1.Service{
+					TypeMeta:   metav1.TypeMeta{Kind: "Service", APIVersion: "core/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: dbName, Namespace: defaultNamespace},
+					Spec: corev1.ServiceSpec{
+						Selector: dbAppLables,
+						Ports: []corev1.ServicePort{
+							{
+								Name:     dbName,
+								Protocol: "TCP",
+								Port:     dbPort,
+							},
+						},
+					},
+				}
+
+				Expect(k8sClient.Create(ctx, &dbService)).Should(Succeed())
+
+				By("deploying artemis")
+
+				peerLabel := "fast-ha-peer"
+				peerPrefix := NextSpecResourceName()
+				ctx := context.Background()
+				peerA := generateArtemisSpec(defaultNamespace)
+				peerA.Name = peerPrefix + "-peer-a"
+
+				peerA.Spec.Acceptors = []brokerv1beta1.AcceptorType{{Name: "tcp", Port: 61616, Expose: true}}
+
+				if !isOpenshift {
+					peerA.Spec.IngressDomain = defaultTestIngressDomain
+				}
+
+				peerA.Spec.Console.Expose = true
+
+				peerA.Spec.DeploymentPlan.PersistenceEnabled = boolFalse
+				peerA.Spec.DeploymentPlan.Clustered = &boolFalse
+
+				peerA.Spec.DeploymentPlan.Labels = map[string]string{peerLabel: peerPrefix}
+
+				peerA.Spec.Env = []corev1.EnvVar{
+					{
+						Name:  "ARTEMIS_EXTRA_LIBS",
+						Value: "/amq/init/config/extra-libs",
+					},
+				}
+
+				By("configuring the broker")
+				peerA.Spec.BrokerProperties = []string{
+					"addressConfigurations.JOBS.routingTypes=ANYCAST",
+					"addressConfigurations.JOBS.queueConfigs.JOBS.routingType=ANYCAST",
+
+					"HAPolicyConfiguration=SHARED_STORE_PRIMARY",
+					"storeConfiguration=DATABASE",
+					"storeConfiguration.jdbcDriverClassName=org.postgresql.Driver",
+					"storeConfiguration.jdbcConnectionUrl=jdbc:postgresql://" + dbName + ":" + fmt.Sprintf("%d", dbPort) + "/postgres?user=postgres&password=postgres",
+				}
+
+				By("patching ss to add init container to download jdbc jar")
+
+				kindMatchSs := "StatefulSet"
+				peerA.Spec.ResourceTemplates = []brokerv1beta1.ResourceTemplate{
+					{
+						Selector: &brokerv1beta1.ResourceSelector{
+							Kind: &kindMatchSs,
+						},
+						Patch: &unstructured.Unstructured{Object: map[string]interface{}{
+							"kind": "StatefulSet",
+							"spec": map[string]interface{}{
+								"template": map[string]interface{}{
+									"spec": map[string]interface{}{
+										"initContainers": []interface{}{
+											map[string]interface{}{
+												"name":  "download-jdbc-driver",
+												"image": version.LatestInitImage,
+												"command": []interface{}{
+													"sh",
+													"-c",
+													"mkdir -p /amq/init/config/extra-libs && curl -o /amq/init/config/extra-libs/postgresql-42.7.1.jar https://jdbc.postgresql.org/download/postgresql-42.7.1.jar",
```

Comment:
in production this would need to come from a container or from pre-populated volume mount. this needs some more investigation. we are not alone!